apps/docs-snippets2/src/wallets/encrypting-and-decrypting-wallets.ts

```diff
@@ -0,0 +1,42 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+// #region full
+// #region encrypting-and-decrypting-json-wallets-1
+import fs from 'fs';
+import { Provider, Wallet } from 'fuels';
+
+import { LOCAL_NETWORK_URL } from '../env';
+
+const provider = await Provider.create(LOCAL_NETWORK_URL);
+
+const wallet = Wallet.generate({ provider });
+
+// Encrypt the wallet
+const password = 'my-password';
+const jsonWallet = await wallet.encrypt(password);
+
+// Save the encrypted wallet to a file
+// e.g. const jsonWallet = fs.writeFileSync('secure-path/my-wallet.json', jsonWallet);
```

Comment:
Is it worthwhile writing + reading to the filesystem - then doing a clean up after the `#endregion`?

Reply:
imo no, a user can determine how they want to do so, this is just a recommendation